presto-clp/src/main/java/com/facebook/presto/plugin/clp/ClpMetadataFilterProvider.java

```
@@ -0,0 +1,221 @@
+ * </ul>
+ * <p>
+ * This provider is used by {@code ClpFilterToKqlConverter} to determine which columns are eligible
+ * for metadata filter push down, and by {@code ClpMySqlSplitProvider} to construct metadata filter
+ * queries that determine which splits to read.
+ */
+public class ClpMetadataFilterProvider
+{
+    private final Map<String, TableConfig> filterMap;
+
+    @Inject
+    public ClpMetadataFilterProvider(ClpConfig config)
+    {
+        requireNonNull(config, "config is null");
+
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            filterMap = mapper.readValue(
+                    new File(config.getMetadataFilterConfig()),
+                    new TypeReference<Map<String, TableConfig>>() {});
+        }
+        catch (IOException e) {
+            throw new PrestoException(CLP_METADATA_FILTER_CONFIG_NOT_FOUND, "Failed to metadata filter config file open.");
+        }
+    }
+
+    public Map<String, TableConfig> getFilterMap()
+    {
+        return filterMap;
+    }
+
+    public void checkContainsAllFilters(SchemaTableName schemaTableName, String metadataFilterKqlQuery)
+    {
+        boolean hasAllMetadataFilterColumns = true;
+        String notFoundFilterColumnName = "";
+        for (String columnName : getFilterNames(format("%s.%s", CONNECTOR_NAME, schemaTableName))) {
+            if (!metadataFilterKqlQuery.contains(columnName)) {
+                hasAllMetadataFilterColumns = false;
+                notFoundFilterColumnName = columnName;
+                break;
+            }
+        }
+        if (!hasAllMetadataFilterColumns) {
+            throw new PrestoException(CLP_MANDATORY_METADATA_FILTER_NOT_VALID, notFoundFilterColumnName + " is a mandatory metadata filter column but not valid");
+        }
+    }
+
+    public String remapFilterSql(String scope, String sql)
+    {
+        String[] splitScope = scope.split("\\.");
+        if (0 == splitScope.length) {
+            return sql;
```

Should we throw an exception here?

---

actually, it never becomes 0😅